cmd/podman/machine/rm.go

```diff
@@ -120,14 +119,12 @@ func rm(_ *cobra.Command, args []string) error {
 	// All actual removal of files and vms should occur after this
 	//
 
-	// TODO Should this be a hard error?
 	if err := providerRm(); err != nil {
-		logrus.Errorf("failed to remove virtual machine from provider for %q", vmName)
+		return fmt.Errorf("failed to remove virtual machine from provider for %q: %v", vmName, err)
```

Comment:
I think the one open question is do we want to call genericRm() even when this failed? I am not yet familiar with the differences between both but my gut tells me yes. So I would recommend to collect/log the errors like done elsewhere.
@baude WDYT?

Reply:
`providerRM` deals with provider specific files. So for QEMU we would remove the QMP Monitor pid file, for example. `genericRM` is where we delete the ignition file, config file, and image. I'm fine with logging the error and then doing a hard error on `genericRM`

Reply:
I would report the error either way but do both.  BTW Ignore ENOTEXIST errors on missing, (You can just logrus.Debug them.)

Reply:
`VMFile.Delete()` ignores ENOTEXIST when reporting the error so all good on that front